torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -583,16 +583,16 @@ def _partition_parameters(
                 self._partition_parameters_cache = [[] for _ in range(self.world_size)]
                 sizes = [0] * self.world_size
                 for param_group in self.param_groups:
-                    params_per_rank: List[List] = [[] for _ in range(self.world_size)]
+                    param_group_params_per_rank: List[List] = [[] for _ in range(self.world_size)]
```

Comment:
This rename is to prevent implicit shadowing of the method's argument `params_per_rank`.